contracts/contracts/FleekAccessControl.sol

```diff
@@ -5,227 +5,178 @@ pragma solidity ^0.8.7;
 import "@openzeppelin/contracts/utils/Counters.sol";
 import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
 
+error MustHaveCollectionRole(uint8 role);
+error MustHaveTokenRole(uint256 tokenId, uint8 role);
+error MustHaveAtLeastOneOwner();
+error RoleAlreadySet();
+
 contract FleekAccessControl is Initializable {
     using Counters for Counters.Counter;
 
-    enum Roles {
-        Owner,
-        Controller
+    /**
+     * @dev All available collection roles.
+     */
+    enum CollectionRoles {
+        Owner
     }
 
-    event TokenRoleGranted(uint256 indexed tokenId, Roles indexed role, address indexed toAddress, address byAddress);
-    event TokenRoleRevoked(uint256 indexed tokenId, Roles indexed role, address indexed toAddress, address byAddress);
-    event CollectionRoleGranted(Roles indexed role, address indexed toAddress, address byAddress);
-    event CollectionRoleRevoked(Roles indexed role, address indexed toAddress, address byAddress);
-
-    struct Role {
-        mapping(address => uint256) indexes;
-        address[] members;
+    /**
+     * @dev All available token roles.
+     */
+    enum TokenRoles {
+        Controller
     }
 
-    Counters.Counter private _collectionRolesVersion;
-    // _collectionRoles[version][role]
-    mapping(uint256 => mapping(Roles => Role)) private _collectionRoles;
+    /**
+     * @dev Emitted when a collection role is granted.
+     */
+    event TokenRoleGranted(
```

Comment:
I think `TokenRoleRevoked` can be merged into this event. `TokenRoleChanged` is better probably.